tests/kafkatest/tests/core/reassign_partitions_test.py

```diff
@@ -86,28 +94,60 @@ def reassign_partitions(self, bounce_brokers):
             self.clean_bounce_some_brokers()
 
         # Wait until finished or timeout
-        wait_until(lambda: self.kafka.verify_reassign_partitions(partition_info), timeout_sec=self.timeout_sec, backoff_sec=.5)
+        wait_until(lambda: self.kafka.verify_reassign_partitions(partition_info),
+                   timeout_sec=self.timeout_sec, backoff_sec=.5)
 
-    @cluster(num_nodes=7)
-    @parametrize(security_protocol="PLAINTEXT", bounce_brokers=True)
-    @parametrize(security_protocol="PLAINTEXT", bounce_brokers=False)
-    def test_reassign_partitions(self, bounce_brokers, security_protocol):
+    def move_start_offset(self):
+        """We move the start offset of the topic by writing really old messages
+        and waiting for them to be cleaned up.
+        """
+        producer = VerifiableProducer(self.test_context, 1, self.kafka, self.topic,
+                                      throughput=-1, enable_idempotence=True,
+                                      create_time=1000)
+        producer.start()
+        wait_until(lambda: producer.num_acked > 0,
+                   timeout_sec=30,
+                   err_msg="Failed to get an acknowledgement for %ds" % 30)
+        # Wait 8 seconds to let the topic be seeded with messages that will
+        # be deleted. The 8 seconds is important, since we should get 2 deleted
+        # segments in this period based on the configured log roll time and the
+        # retention check interval.
+        time.sleep(8)
+        producer.stop()
+        self.logger.info("Seeded topic with %d messages which will be deleted" %\
+                         producer.num_acked)
+        # Since the configured check interval is 5 seconds, we wait another
+        # 6 seconds to ensure that at least one more cleaning so that the last
+        # segment is deleted.
+        time.sleep(6)
```

Comment:
Alternatively, we could poll the start offset and wait for it to match the end offset.

Reply:
Yes. That was my plan B, if the test in it's current form was flakey. But that would be a bit of extra work since this information is not easily exposed by the current console consumer or verifiable consumer.

But since we got 120 successful runs of this test, the current timeouts are sufficient IMO.

Reply:
Fair enough. A bit annoying sometimes that we don't have the full APIs to work with.

Reply:
I added a comment to the code to mention the alternate, more robust, way to do this. We can execute on it if/when the test starts flaking out.